core/src/memory.rs

```diff
@@ -578,6 +837,44 @@ mod tests {
             })
         }
 
+        fn wasm_pages_amount() -> impl Strategy<Value = WasmPagesAmount> {
+            wasm_pages_amount_with_range(0, u16::MAX as u32 + 1)
+        }
+
+        // This high-order strategy generates valid memory parameters in a specific way that allows passing `AllocationContext::validate_memory_params` checks.
+        fn combined_memory_params() -> impl Strategy<
+            Value = (
```

Comment:
Please append new struct here `MemoryParams`

Reply:
I have a similar idea, but it seems impossible to create a combined strategy that uses a `struct` for `Value` instead of a tuple. The `prop_flat_map` combinator transforms a tuple of strategies into a strategy of tuples, but it cannot do the same for a struct.

Reply:
Hm, yes. May be the most correct way is to implement `Arbitrary` for `struct MemoryParams`.
You can see examples in common/numerated/src/mock.rs. But may be better to make it in other PR

Reply:
Created issue #3906 3906